python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr.py

```diff
@@ -250,9 +254,12 @@ def _read_application_logs(
             bucket = logs_uri.split("/")[2]
             prefix = "/".join(logs_uri.split("/")[3:])
 
-            # discover container (application) logs (e.g. Python logs) and forward all of them
-            # ex. /containers/application_1727881613116_0001/container_1727881613116_0001_01_000001/stdout.gz
-            containers_prefix = os.path.join(prefix, f"{cluster_id}/containers/")
+            # discover application logs (e.g. Python logs) and forward all of them
+            # ex. /containers/application_1727881613116_0001/container_1727881613116_0001_01_000001/stdout.gz for Yarn
+            # or /steps/... for EMR steps
+            application_logs_prefix = os.path.join(
+                prefix, f"{cluster_id}/{self.s3_application_logs_prefix or 'containers'}/"
```

Comment:
nit: why not set the default value to `containers` @ [1]? feel like thats marginally more straight forward given the doc block

Reply:
Agree. Not sure what made me use a default `None` value.